src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,53 @@
-import React from 'react';
+/* eslint-disable react/button-has-type */
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { CommentData } from '../types/Comment';
+
+interface NewCommentFormProps {
+  commentAdd: (comment: CommentData) => Promise<void>,
+}
+
+export const NewCommentForm: React.FC<NewCommentFormProps> = ({
+  commentAdd,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+  const [commentLoading, setIsCommentLoading] = useState(false);
+  const [validSubmit, setValidSubmit] = useState(true);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (name && email && body) {
+      const newComment = {
+        name,
+        email,
+        body,
+      };
+
+      setIsCommentLoading(true);
+      setValidSubmit(true);
+
+      commentAdd(newComment)
+        .then(() => setBody(' '))
+        .finally(() => setIsCommentLoading(false));
+    } else {
+      setValidSubmit(false);
+    }
+  };
+
+  const handleFormClear = () => {
+    setName('');
+    setEmail('');
+    setBody('');
+    setValidSubmit(true);
+  };
+
+  const validInput = (inputType: string) => !validSubmit && !inputType;
```

Comment:
The validInput function should return true if the input is valid, so the condition should be !validSubmit && inputType.